### PR TITLE
Pass in initial admin password and remove admin:admin references

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
Starting with 2.12.0, security plugin requires an initial admin password to be set via env. variable when installing the demo configuration. This PR is to update documentation and CI to accommodate that change.

### Issues Resolved
Closes: https://github.com/opensearch-project/opensearch-py/issues/759

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by saimedhi** [comment]
Hello @derek-ho,

Is this pull request ready for review? If there's any context I might be missing, could you share details on why this implementation is necessary or guide me to the relevant issue?

Thank you :)

**Comment by derek-ho** [comment]
Some logic may need to be added after 2.12.0 release in test files like here: https://github.com/opensearch-project/opensearch-py/blob/main/opensearchpy/helpers/test.py#L37 to replace this with new password conditionally, based on the version the CI is running against.

**Comment by derek-ho** [comment]
@saimedhi @VachaShah this PR is ready for review now - I am not sure what approach you folks think is appropriate. I think this PR should be able to be merged as is, but after 2.12.0 I would expect some tests to fail for 2.12.0 release, such as here: https://github.com/opensearch-project/opensearch-py/blob/main/opensearchpy/helpers/test.py#L37. What do you folks think?

**Comment by saimedhi** [comment]
> @saimedhi @VachaShah this PR is ready for review now - I am not sure what approach you folks think is appropriate. I think this PR should be able to be merged as is, but after 2.12.0 I would expect some tests to fail for 2.12.0 release, such as here: https://github.com/opensearch-project/opensearch-py/blob/main/opensearchpy/helpers/test.py#L37. What do you folks think?

If it's not urgent, we can wait until the 2.12.0 release to make the code changes. This allows us to align with the latest release. @VachaShah what do you think?

**Comment by VachaShah** [comment]
I am wondering why we can't make all the changes now since we already [test](https://github.com/opensearch-project/opensearch-py/blob/main/.github/workflows/integration-unreleased.yml#L11) this client against the developing 2.x line. @derek-ho I believe you should be able to add the version checks in as well and test the code out. LMK if I am missing something here.

**Comment by derek-ho** [comment]
> > @saimedhi @VachaShah this PR is ready for review now - I am not sure what approach you folks think is appropriate. I think this PR should be able to be merged as is, but after 2.12.0 I would expect some tests to fail for 2.12.0 release, such as here: https://github.com/opensearch-project/opensearch-py/blob/main/opensearchpy/helpers/test.py#L37. What do you folks think?
> 
> If it's not urgent, we can wait until the 2.12.0 release to make the code changes. This allows us to align with the latest release. @VachaShah what do you think?

I am ok with waiting for now, if the regular CI is not broken. I will turn this PR to draft for now.

**Comment by dblock** [comment]
> @VachaShah I believe those tests are run against 2.x OpenSearch **without** security, so thus we are not seeing those fail yet.

Are you able to help enable that?

**Comment by DarshitChanpura** [comment]
Now that 2.12 is released this should be unblocked. Would the maintainers review and merge this?

**Comment by VachaShah** [comment]
@derek-ho Is this ready for review?

**Comment by dblock** [comment]
@DarshitChanpura also needs a rebase, please

**Comment by DarshitChanpura** [comment]
@derek-ho Would you please rebase this and mark as ready for review? I don't have write access to push to your branch.

**Comment by dblock** [comment]
tests without security are still failing

**Comment by saimedhi** [comment]
@derek-ho Please fix these failing tests https://github.com/opensearch-project/opensearch-py/actions/runs/8072055726/job/22087332746

**Comment by derek-ho** [comment]
> @derek-ho Please fix these failing tests https://github.com/opensearch-project/opensearch-py/actions/runs/8072055726/job/22087332746

I am pretty sure the without security tests are not because of this change - I see them failing on other PRs as well. I am working on fixing the 2.12 with security test case, but running into issues with extracting version information within the test, as it seems to only be available outside of the test case. I will try a bit longer, but may need maintainers help for that.

**Comment by saimedhi** [comment]
> > @derek-ho Please fix these failing tests https://github.com/opensearch-project/opensearch-py/actions/runs/8072055726/job/22087332746
> 
> I am pretty sure the without security tests are not because of this change - I see them failing on other PRs as well. I am working on fixing the 2.12 with security test case, but running into issues with extracting version information within the test, as it seems to only be available outside of the test case. I will try a bit longer, but may need maintainers help for that.

Yes, other failures are unrelated to this PR. Apologies for any confusion caused. I specifically referred to version (2.12.0, true) as seen here: [link](https://github.com/opensearch-project/opensearch-py/actions/runs/8072055726/job/22087332746). Thank you :)

**Comment by saimedhi** [comment]
Hello @derek-ho,

Could you please take a look at finishing this PR when you have a moment?

Thanks a lot!

**Comment by saimedhi** [comment]
Hello @derek-ho, could you please take a look at this PR and let me know if it's ready for review? opensearch-py currently lacks compatibility with the latest OpenSearch versions 2.12.0, 2.13.0, and 2.14.0. It would be great if we could resolve this issue and get the PR merged soon. Are you facing any difficulties or need assistance in this regard?

File: .ci/run-opensearch.sh
Comment (dblock) [inline .ci/run-opensearch.sh:66]: This won't work for 2.13, will it?

Comment (DarshitChanpura) [reply .ci/run-opensearch.sh:66]: It should since admin credentials changes were released in 2.12. If you are referencing script execution in general, it works as expected returning `admin:admin` for versions <2.12 and `admin:myStronPassword123!` for those >= 2.12

**Review by DarshitChanpura — APPROVED** [review]

File: .ci/run-opensearch.sh
Comment (dblock) [reply .ci/run-opensearch.sh:66]: Ok, but wouldn't it be simpler to just set `OPENSEARCH_USERNAME` and `OPENSEARCH_PASSWORD` in `env:` in the workflow?

**Review by dblock — APPROVED** [review]

File: .ci/run-opensearch.sh
Comment (DarshitChanpura) [reply .ci/run-opensearch.sh:66]: yes, it should be, however on line 64 we make a status check call that would require the correct credentials to be passed.

**Comment by dblock** [comment]
@derek-ho I'm good with what's here if you can get it to green. 

Without trying it, I think we can specify the username/password in .github workflow and remove all the logic from the .sh scripts. We can also do this later.

**Comment by derek-ho** [comment]
Agree @dblock there is probably a simpler way to do it, but I am not an expert in Python and just focused on getting it to a good state that we can start testing against the newer versions. It can be cleaned up in a follow up, but I may need assistance for that part.

**Comment by dblock** [comment]
> Agree @dblock there is probably a simpler way to do it, but I am not an expert in Python and just focused on getting it to a good state that we can start testing against the newer versions. It can be cleaned up in a follow up, but I may need assistance for that part.

I'm good with whatever works. We can improve later.

File: .ci/run-repository.sh
Comment (derek-ho) [reply .ci/run-repository.sh:62]: Yea, we can skip it but I was having trouble reading the env correctly within the testing code itself to determine which credentials to use when setting up the connection. It seems that I needed to set the env. in the sh file right before the tests ran for it to work. Not sure if there is a simpler way.

**Comment by derek-ho** [comment]
@dblock the CI seems to pass with the latest changes, can we merge this as is? Do you want me to create a follow up issue to clean up the logic/code for CI?

**Comment by derek-ho** [comment]
Created follow up issue here: https://github.com/opensearch-project/opensearch-py/issues/764

**Review by dblock — APPROVED** [review]

**Pull request merged**
